src/Controllers/MapController.ts

```diff
@@ -902,75 +905,45 @@ module StreamStats.Controllers {
+                                    gtag('event', 'ValidatePoint',{ 'Label': 'Not advised (no point query)' });
+                                } else { // If there are no warning messages; there was an exclusion polygon layer, and the pourpoint was valid 
+                                    this.toaster.pop("success", "Your clicked point is valid", "Delineating your basin now...", 5000)
+                                    gtag('event', 'ValidatePoint',{ 'Label': 'Valid' });
+                                }
                                 this.startDelineate(latlng, false);
-                            }
-
-                            //otherwise parse exclude Codes
-                            else {
+                            } else if (result.inExclude == true) { // If point is in exclusion polygon
                                 this.studyArea.checkingDelineatedPoint = false;
-                                var excludeCode = results.features[0].properties.ExcludeCod;
-                                var popupMsg = results.features[0].properties.ExcludeRea;
-                                if (excludeCode == 1) {
-                                    this.toaster.pop("error", "Delineation and flow statistic computation not allowed here", popupMsg, 0);
-                                    //ga event
+                                if (result.type == 1) { // If point is in a hard exclusion polygon (no delineation allowed)
+                                    // Prohibit delineation
+                                    this.toaster.pop("error", "Delineation and flow statistic computation not allowed here", result.message.text, 0);
                                     gtag('event', 'ValidatePoint',{ 'Label': 'Not allowed' });
-                                }
-                                else {
-                                    this.toaster.pop("warning", "Delineation and flow statistic computation possible but not advised", popupMsg, true, 0);
-                                    this.startDelineate(latlng, true, popupMsg);
-                                    //ga event
+                                } else { // If point is in a soft exclusion polygon (delineation allowed)
+                                    // Warn the user
+                                    this.toaster.pop("warning", "Warning", result.message.text, true, 0);
                                     gtag('event', 'ValidatePoint',{ 'Label': 'Not advised' });
+                                    this.startDelineate(latlng, true, result.message.text);
                                 }
+                                // TODO in future issue: add more types of exlusion polygons
                             }
-
-                            this.cursorStyle = 'pointer';
+                            
                         });

```

Comment:
```suggestion
                        }, (error) => {
                            //toast message when error              
                        }).finally(() => {  
                            // probably don't need anything in here      
                        });

```

Reply:
If the exclude poly service fails in the BP, we return the point as if it is not in an exclude polygon and proceed with delineation, so I'll plan to replicate that functionality

Reply:
The `}, (error) => {` line wasn't triggering with my testing, so I also checked the status of the response to catch when the query exclude poly service was failing.

I tested that by changing excludePolygonQuery to this (erroneous POST body):

```
var excludePolygonQuery = {
                "regionXYZ": region,
                "points": [
                    {
                    "lat": lat,
                    "long": lng
                    }
                ]
            }
```